documentation/modules/cruise-control/proc-implementing-rebalance-proposal.adoc

```diff
@@ -0,0 +1,73 @@
+// Module included in the following assemblies:
+//
+// assembly-cruise-control-concepts.adoc
+
+[id='proc-implementing-optimization-proposal-{context}']
+
+= Implementing an optimization proposal
+
+You can implement an optimization proposal with a status of `ProposalReady`.
+Cruise Control will then reassign partitions to brokers as summarized in the optimization proposal.
+
+WARNING: This is not a dry run. 
+Review the optimization proposal before you implement it.
+
+.Prerequisites
+
+* You have xref:proc-generating-optimization-proposals-{context}[generated an optimization proposal].
+
+* The optimization proposal is ready to implement (the `KafkaRebalance` custom resource status is `ProposalReady`).
+
+.Procedure
+
+Perform these steps for the optimization proposal that you want to implement:
+
+. If the `KafkaRebalance` status has been `ProposalReady` for some time, refresh the optimization proposal in case it has become out of date:
+
+.. Annotate the `KafkaRebalance` resource in Kubernetes:
++
+[source,shell,subs="+quotes"]
+----
+kubectl annotate kafkarebalance _cluster-name_-kafka strimzi.io/rebalance=refresh
+----
+
+.. Check the status of the `KafkaRebalance` resource:
++
+[source,shell,subs="+quotes"]
+----
+kubectl get kafka kafkarebalance -o jsonpath='{.status}'
+----
+
+.. Wait until the status changes to `ProposalReady`.
+
+. Approve the optimization proposal that you want to implement.
++
+Annotate the `KafkaRebalance` resource in Kubernetes:
++
+[source,shell,subs="+quotes"]
+----
+kubectl annotate kafkarebalance _cluster-name_-kafka strimzi.io/rebalance=approved
+----
+
+. The Cluster Operator detects the annotated resource and instructs Cruise Control to rebalance the Kafka cluster.
+
+. Check the status of the `KafkaRebalance` resource:
++
+[source,shell,subs="+quotes"]
+----
+kubectl get kafka kafkarebalance -o jsonpath='{.status}'
+----
+
+. Cruise Control returns one of three statuses:
+
+** Rebalancing: The cluster rebalance operation is in progress. 
```

Comment:
I may have missed it, but where is the procedure for stopping an ongoing rebalance?

Reply:
I decided to leave that out because it seemed like an edge case. Should I add one?

Reply:
Yes definitely! Rebalances can go on for hours and seriously effect a cluster's performance. A user may mis-configure a rebalance request and not be paying attention to the summary and only realise they have made a mistake once it has begun. We need to give them an escape hatch!